File: Hot 100/169. 多数元素.go
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        major = 0
        count = 0
        
        for n in nums:
            if count == 0:
                major = n
            if n == major:
                count = count + 1
            else:
                count = count - 1

        return major

File: CodeTop/169. 多数元素.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int candidate = -1;
        int count = 0;
        for(auto num : nums) {
            if(num == candidate) {
                count++;
            } else if(count == 0) {
                candidate = num;
            } else {
                count--;
            }
        }
        return candidate;
    }
};